spec/models/service_template_ansible_playbook_spec.rb

```diff
@@ -0,0 +1,106 @@
+            :credential_id         => auth_one.id,
+            :network_credential_id => auth_two.id,
+            :playbook_id           => playbook.id,
+          },
+        }
+      }
+    end
+
+    let(:catalog_item_options_two) do
+      {
+        :name        => 'playbook service',
+        :display     => 'false',
+        :description => 'a description',
+        :config_info => {
+          :provision   => {
+            :new_dialog_name => 'playbook dialog',
+            :playbook_id     => 1,
+            :extra_vars      => {
+              'key1' => 'val1',
+              'key2' => 'val2'
+            }
+          },
+          :reconfigure => {
+            :new_dialog_name => 'playbook dialog reconfigure',
+            :playbook_id     => 5,
+          },
+          :retirement  => {
+            :new_dialog_name => 'playbook dialog retirement',
+            :playbook_id     => 3,
+          },
+        }
+      }
+    end
+
+    it '#create_job_templates' do
+      expect(described_class).to receive(:create_job_template).exactly(3).times.and_return(job_template)
+      options_hash = described_class.create_job_templates(catalog_item_options_two[:name], catalog_item_options_two[:description], catalog_item_options_two[:config_info], 'system')
+      [:provision, :retirement, :reconfigure].each do |action|
+        expect(options_hash[action.to_sym][:configuration_template]).to eq job_template
+      end
+    end
+
+    it '#create_job_template' do
+      expect(described_class).to receive(:build_parameter_list).and_return([ems, {}])
+      expect(ManageIQ::Providers::AnsibleTower::AutomationManager::ConfigurationScript).to receive(:create_in_provider_queue).once.with(ems.id, {}, 'system')
+      expect(MiqTask).to receive(:wait_for_taskid).with(any_args).once.and_return(instance_double('MiqTask', :task_results => {}, :status => 'Ok'))
+
+      described_class.create_job_template(catalog_item_options[:name], catalog_item_options[:description], catalog_item_options[:config_info], 'system')
+    end
+
+    it 'create_job_template exception' do
+      expect(described_class).to receive(:build_parameter_list).and_return([ems, {}])
+      expect(ManageIQ::Providers::AnsibleTower::AutomationManager::ConfigurationScript).to receive(:create_in_provider_queue).once.with(ems.id, {}, 'system')
+      expect(MiqTask).to receive(:wait_for_taskid).with(any_args).once.and_raise(Exception, 'bad job template')
+
+      expect { described_class.create_job_template(catalog_item_options[:name], catalog_item_options[:description], catalog_item_options[:config_info], 'system') }.to raise_error(Exception)
+    end
+
+    it '#build_parameter_list' do
+      name = catalog_item_options[:name]
+      description = catalog_item_options[:description]
+      info = catalog_item_options[:config_info][:provision]
+      _tower, params = described_class.build_parameter_list(name, description, info)
+
+      expect(params[:name]).to eq name
```

Comment:
This could be a single expectation like:
```ruby
expect(params).to have_attributes(
  :name => name,
  :description => description,
...etc...
)
```